Dockerfile

```diff
@@ -0,0 +1,56 @@
+FROM debian:stable as builder
+
+RUN apt-get update && apt-get install -y --no-install-recommends curl git wget ca-certificates
+
+WORKDIR /java/
+
+# Download the JRE for copying to the image to run the OTP server
+RUN curl -Lo jre17.tar.gz https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.7%2B7/OpenJDK17U-jre_x64_linux_hotspot_17.0.7_7.tar.gz
+RUN tar xvf jre17.tar.gz && rm jre17.tar.gz
```

Comment:
Why not download it directly to the runner image? The current process of downloading first and then copying seems a bit wasteful.

Reply:
I think the way it's doing this is ideal actually. Since the download happens as one of the first steps and the file won't ever change, it can basically be cached forever (though this mostly just makes a difference locally). But also I think the COPY is a relatively cheap operation, and I like having all of the dependencies that need to get copied over to the final image specified in a single place.

Reply:
Good point about caching and also good point about having dependencies together. If it doesn't take much extra time (I haven't measured it), then I guess it's fine.